### PR TITLE
update to use lable selector for test cases

### PR DESCRIPTION
**Pull request by hchenxa**
Signed-off-by: hchenxa <huichen@redhat.com>

https://github.com/open-cluster-management/backlog/issues/14099

### CODE REVIEW

File: pkg/tests/observability_alert_test.go [hunk 1]
```diff
@@ -89,10 +93,11 @@ var _ = Describe("Observability:", func() {
...
 	It("[P1][Sev1][Observability][Stable] Should have the alertmanager configured in rule (alert/g0)", func() {
 		By("Checking if --alertmanagers.url or --alertmanager.config or --alertmanagers.config-file is configured in rule")
-		name := MCO_CR_NAME + "-thanos-rule"
-		rule, err := hubClient.AppsV1().StatefulSets(MCO_NAMESPACE).Get(name, metav1.GetOptions{})
+		rules, err := hubClient.AppsV1().StatefulSets(MCO_NAMESPACE).List(metav1.ListOptions{
+			LabelSelector: THANOS_RULE_LABEL,
+		})
 		Expect(err).NotTo(HaveOccurred())
-		argList := rule.Spec.Template.Spec.Containers[0].Args
+		argList := (*rules).Items[0].Spec.Template.Spec.Containers[0].Args
```
Comment (morvencao) [inline pkg/tests/observability_alert_test.go:100]: Should we check if the length of `Items` is `>= 1`?

File: pkg/tests/observability_retention_test.go [hunk 2]
```diff
@@ -61,12 +61,13 @@ var _ = Describe("Observability:", func() {
...
-			name := MCO_CR_NAME + "-thanos-compact"
-			compact, err := hubClient.AppsV1().StatefulSets(MCO_NAMESPACE).Get(name, metav1.GetOptions{})
+			compacts, err := hubClient.AppsV1().StatefulSets(MCO_NAMESPACE).List(metav1.ListOptions{
+				LabelSelector: THANOS_COMPACT_LABEL,
+			})
 			if err != nil {
 				return err
 			}
-			argList := compact.Spec.Template.Spec.Containers[0].Args
+			argList := (*compacts).Items[0].Spec.Template.Spec.Containers[0].Args
```
Comment (morvencao) [inline pkg/tests/observability_retention_test.go:70]: same here.

File: pkg/tests/observability_retention_test.go [hunk 3]
```diff
@@ -79,12 +80,13 @@ var _ = Describe("Observability:", func() {
...
-			name := MCO_CR_NAME + "-thanos-store-shard-0"
-			store, err := hubClient.AppsV1().StatefulSets(MCO_NAMESPACE).Get(name, metav1.GetOptions{})
+			stores, err := hubClient.AppsV1().StatefulSets(MCO_NAMESPACE).List(metav1.ListOptions{
+				LabelSelector: THANOS_STORE_LABEL,
+			})
 			if err != nil {
 				return err
 			}
-			argList := store.Spec.Template.Spec.Containers[0].Args
+			argList := (*stores).Items[0].Spec.Template.Spec.Containers[0].Args
```
Comment (morvencao) [inline pkg/tests/observability_retention_test.go:89]: and here

File: pkg/tests/observability_retention_test.go [hunk 4]
```diff
@@ -115,12 +118,13 @@ var _ = Describe("Observability:", func() {
...
-			name := MCO_CR_NAME + "-thanos-rule"
-			rule, err := hubClient.AppsV1().StatefulSets(MCO_NAMESPACE).Get(name, metav1.GetOptions{})
+			rules, err := hubClient.AppsV1().StatefulSets(MCO_NAMESPACE).List(metav1.ListOptions{
+				LabelSelector: THANOS_RULE_LABEL,
+			})
 			if err != nil {
 				return err
 			}
-			argList := rule.Spec.Template.Spec.Containers[0].Args
+			argList := (*rules).Items[0].Spec.Template.Spec.Containers[0].Args
```
Comment (morvencao) [inline pkg/tests/observability_retention_test.go:127]: also this

File: pkg/tests/observability_retention_test.go [hunk 5]
```diff
@@ -133,12 +137,13 @@ var _ = Describe("Observability:", func() {
...
```
Comment (morvencao) [inline pkg/tests/observability_retention_test.go:146]: this one also.

**Review by morvencao — COMMENTED** [review]
others look good to me.

File: pkg/tests/observability_alert_test.go (hunk 1 above)
Comment (hchenxa) [reply pkg/tests/observability_alert_test.go:102]: add `Expect(len(rules.Items)).NotTo(Equal(0))`

**Pull request merged**